src/wrapper/Plumed.h

```diff
@@ -2874,28 +2955,258 @@ Plumed(Plumed&&p)__PLUMED_WRAPPER_CXX_NOEXCEPT :
+    static_assert(sizeof(T)%value_size==0,"custom array has incorrect size");
+    auto newptr=reinterpret_cast<value_type*>(val);
+    auto newnelem=nelem*sizeof(T)/sizeof(value_type);
+    cmd_with_nelem(key,newptr,newnelem);
+  }
+
+/// cmd_helper_with_nelem with pointer to simple type val.
+  template<typename T, typename std::enable_if<!wrapper::is_custom_array<T>::value, int>::type = 0>
+  void cmd_with_nelem(const char*key,T* val, __PLUMED_WRAPPER_STD size_t nelem) {
+    // pointer or value, directly managed by SafePtr
+    SafePtr s(val,nelem,nullptr);
+    cmd_priv(main,key,&s);
+  }
+public:
+
+  /**
+     Send a command to this plumed object
+      \param key The name of the command to be executed
+      \param val The argument.
+      \note This overload detects temporaries
+  */
+  template<typename T>
+  void cmd(const char*key,T&& val) {
+    cmd_helper(key,std::forward<T>(val),true);
+  }
+
+  /**
+     Send a command to this plumed object
+      \param key The name of the command to be executed
+      \param val The argument.
+      \note This overload detect non temporaries C array references.
+  */
+  template<typename T, typename std::enable_if<wrapper::is_array<T>::value, int>::type = 0>
+  void cmd(const char*key,T& val) {
+    std::size_t shape[] { wrapper::is_array<T>::size, 0 };
+    cmd_helper_with_shape(key,reinterpret_cast<typename wrapper::is_array<T>::value_type*>(val),shape);
+  }
+
+  /**
+     Send a command to this plumed object
+      \param key The name of the command to be executed
+      \param val The argument.
+      \note This overload detect non temporaries non C array references.
+  */
+  template<typename T, typename std::enable_if<!wrapper::is_array<T>::value, int>::type = 0>
+  void cmd(const char*key,T& val) {
+    cmd_helper(key,std::forward<T>(val));
+  }
+
+  /**
+     Send a command to this plumed object
+      \param key The name of the command to be executed
+      \param val The argument.
+      \note This overload accepts a pointer and corresponding size
+            information. It's usage is discouraged:
+            the overload accepting shape information should be preferred.
+  */
+
+  template<typename T>
+  void cmd(const char*key,T* val, __PLUMED_WRAPPER_STD size_t nelem) {
+#if __PLUMED_WRAPPER_CXX_DETECT_SHAPES_STRICT
+    static_assert("in strict mode you cannot pass nelem, please pass full shape instead");
+#endif
+    cmd_with_nelem(key,val,nelem);
+  }
+
+
+  /**
+     Send a command to this plumed object
+      \param key The name of the command to be executed
+      \param val The argument.
+      \note This overload accepts a pointer and corresponding shape
+            information. Shape is passed a size_t pointer,
+            but the overload accepting an initializer_list
+            has a more friendly syntax.
+  */
+  template<typename T>
+  void cmd(const char*key,T* val, __PLUMED_WRAPPER_STD size_t* shape) {
```

Comment:
I think the problem may raise here, the other cmd with shape sainitizes via `make_shape`
Maybe using an ad hoc modified `make_shape` also here may sanitize the input also here